### PR TITLE
Webpack DLL

### PR DESCRIPTION
**Pull request by zigzag95**
Usage:
npm run server:withdll

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -12,8 +12,8 @@ import { AppState } from './app.service';
 @Component({
   selector: 'app',
   encapsulation: ViewEncapsulation.None,
-  styleUrls: [
-    './app.style.css'
+  styles: [
+    require('./app.style.css')
```
Comment (katallaxie) [inline src/app/app.component.ts]: `@angularclass/conventions-loader` is doing this for you 🍶

File: src/app/home/home.component.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ import { XLarge } from './x-large';
     Title
   ],
   // Our list of styles in our component. We may add more to compose many styles together
-  styleUrls: [ './home.style.css' ],
+  styles: [ require('./home.style.css') ],
```
Comment (katallaxie) [inline src/app/home/home.component.ts]: @angularclass/conventions-loader

File: package.json [hunk 3]
```diff
@@ -11,6 +11,8 @@
   "homepage": "https://github.com/angularclass/angular2-webpack-starter",
   "license": "MIT",
   "scripts": {
+    "dll": "webpack --config src/webpack.dll.ts --env.ENV development",
```
Comment (katallaxie) [inline package.json]: Shouldn't we make this default?

**Comment by katallaxie** [comment]
Why is there the `index-dll.html`? Shouldn't we make this default behavior.

**Comment by d3viant0ne** [comment]
Given @gdi2290 is already implementing this in the Angular2 seed, would it not make more sense to simply port that over to the webpack starter. Consistency ftw :)

File: src/custom-typings.d.ts [hunk 4]
```diff
@@ -121,3 +121,36 @@ interface ErrorConstructor extends ErrorStackTraceLimit {}
 interface NodeRequireFunction extends Es6PromiseLoader  {}
 interface NodeModule extends WebpackModule {}
 interface Global extends GlobalEnvironment  {}
+
+
+/*
+ * Custom Type Definitions
+ * When including 3rd party modules you also need to include the type definition for the module
+ * if they don't provide one within the module. You can try to install it with typings
+typings install node --save
```
Comment (yuanfeiz) [inline src/custom-typings.d.ts]: For typescript 2.0, the old typings system might not be required.

**Comment by d3viant0ne** [comment]
Closing - The source fork no longer exists

**Pull request closed without merging**